shell/models/__tests__/management.cattle.io.node.ts

```diff
@@ -0,0 +1,82 @@
+    resetMocks();
+  });
+  it('should return addresses if they are present directly on the resource status', () => {
+    const mgmtNode = new MgmtNode({ status: { addresses: ['foo'] } }, ctx);
+
+    expect(mgmtNode.addresses).toStrictEqual(['foo']);
+    resetMocks();
+  });
+  it('should return addresses if they are not present directly on the resource status but are on "status.internalNodeStatus"', () => {
+    const mgmtNode = new MgmtNode({ status: { internalNodeStatus: { addresses: ['bar'] } } }, ctx);
+
+    expect(mgmtNode.addresses).toStrictEqual(['bar']);
+    resetMocks();
+  });
+  it('should return an internalIp if addresses includes an object with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { addresses: [{ type: 'InternalIP', address: 'foo' }] } }, ctx);
+
+    expect(mgmtNode.internalIp).toStrictEqual('foo');
+    resetMocks();
+  });
+  it('should return an externalIp if addresses includes an object with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { addresses: [{ type: 'ExternalIP', address: 'bar' }] } }, ctx);
+
+    expect(mgmtNode.externalIp).toStrictEqual('bar');
+    resetMocks();
+  });
+  it('should return an internalIp if internalNodeStatus.addresses includes an object with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { internalNodeStatus: { addresses: [{ type: 'InternalIP', address: 'foo' }] } } }, ctx);
+
+    expect(mgmtNode.internalIp).toStrictEqual('foo');
+    resetMocks();
+  });
+  it('should return an externalIp if internalNodeStatus.addresses includes an object with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { internalNodeStatus: { addresses: [{ type: 'ExternalIP', address: 'bar' }] } } }, ctx);
+
+    expect(mgmtNode.externalIp).toStrictEqual('bar');
+    resetMocks();
+  });
+  it('should return an internalIp if addresses and internalNodeStatus.addresses do not provide an internal ip and the status includes an rkeNode key with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { rkeNode: { internalAddress: 'foo' } } }, ctx);
+
+    expect(mgmtNode.internalIp).toStrictEqual('foo');
+    resetMocks();
+  });
+  it('should return an externalIp if addresses and internalNodeStatus.addresses do not provide an external ip and the status includes an rkeNode key with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { rkeNode: { address: 'bar' } } }, ctx);
+
+    expect(mgmtNode.externalIp).toStrictEqual('bar');
+    resetMocks();
+  });
+  it('should return an appropriate message if there is no internalIp to display', () => {
+    const mgmtNode = new MgmtNode({ status: {} }, ctx);
+
+    expect(mgmtNode.internalIp).toStrictEqual('baz');
+    expect(t).toHaveBeenCalledTimes(1);
+    expect(t).toHaveBeenCalledWith('generic.none');
```

Comment:
Where is `generic.none` coming from?
It also should be a separated test, so it would explain what is that for and what you are trying to test. Currently is not clear.

Reply:
`generic.none` is coming from the model code itself, it's the translation string that gets fed to `t` if the model getter for internalIp or externalIp don't return an address. This code is part of this model so it belongs in this test suite. This test doesn't touch the external method `t` but ensures that the code in the model is calling it with the expected params given the right circumstances which is exactly what these unit tests are for.